src/Microsoft.Framework.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -9,23 +9,8 @@ public interface IConfiguration
     {
         string this[string key] { get; set; }
 
-        /// <summary>
-        /// 
-        /// </summary>
-        /// <param name="key">A case insensitive name.</param>
-        /// <returns>The value associated with the given key, or null if none is found.</returns>
-        string Get(string key);
+        IConfigurationSection GetSection(string key);
 
-        bool TryGet(string key, out string value);
-
-        IConfiguration GetConfigurationSection(string key);
-
-        IEnumerable<KeyValuePair<string, IConfiguration>> GetConfigurationSections();
-
-        IEnumerable<KeyValuePair<string, IConfiguration>> GetConfigurationSections(string key);
-
-        void Set(string key, string value);
-
-        void Reload();
+        IEnumerable<IConfigurationSection> GetChildren();
```

Comment:
I like `FindSection(key) so I would be OK with `FindSection(key)`/`GetChildren()`, but it would need to go through review again with a number of folks, so I don't think we shoukd change it right now. Can you file a bug?